src/index.js

```diff
@@ -3,45 +3,26 @@
 //And if you haven't heard of Daniel Shiffman, watch his stuff. It is either
 //done in processing or p5.js of which one is a port of the other
 
-//Add scenes from scene files here
-const scenes = {
-    "Challenge Mode": challengeMode, ///src/scenes/challengeMode.js
-    "Cheat Sheet": cheatSheet, ///src/scenes/cheatSheet.js
-    "Game Over": gameOver, ///src/scenes/gameOver.js
-    "Instructions": instructions, ///src/scenes/instructions.js
-    "Main Menu": mainMenu, ///src/scenes/mainMenu.js
-    "Practice Mode": practiceMode, ///src/scenes/practiceMode.js
-    "Settings": settings, ///src/scenes/settings.js
-    "Level Countdown": levelCountdown ///src/scenes/levelCountdown.js
-};
+import sceneManager from './objects/sceneManager.js';
+import { MAIN_MENU } from './constants/sceneConstants.js';
+import canvasManager from './objects/canvasManager.js';
 
-//Change this to change the scene between the options above
-let currentScene = "Main Menu";
-
-//dimensions of the canvas in pixels
-const height = 480;
-const width = 640;
-
-//P5.JS setup function
-function setup()
-{
-    createCanvas(width,height);
-    scenes[currentScene]["setup"]();
-}
-
-//P5.JS draw function
-//This gets called every frame
-//Unless there is something global, the draw functions should be changed in
-//their respective files (as referenced above).
-function draw()
+/**
+ * Our index.js file serves as the gateway into the rest of our application,
+ * and this setup method hooks into the p5 setup function, which is the gateway to all
+ * things p5 in our application.
+ */
+window.setup = function()
 {
-    scenes[currentScene]["draw"]();
-    imageButtons.forEach((e, i, a) => e.show());
+    canvasManager.createCanvas();
+    sceneManager.setCurrentScene(MAIN_MENU);
```

Comment:
These changes makes our `index.js` file more of an orchestrator, as the specific implementation details of how things are set up have been moved into their respective classes; this file no longer needs to know _how_ to set things up, just who to call to get the work done.